library/manageiq_custom_attributes.py

```diff
@@ -0,0 +1,254 @@
+        entities_list = getattr(self.client.collections, ManageIQ.supported_entities[entity_type])
+        return next((e.id for e in entities_list if e.name == entity_name), None)
+
+    def get_entity_cas(self, entity_type, entity_id):
+        """ Returns the entity's custom attributes
+        """
+        try:
+            url = '{api_url}/{entity_type}/{id}?expand=custom_attributes'.format(
+                   api_url=self.api_url,
+                   entity_type=ManageIQ.supported_entities[entity_type],
+                   id=entity_id)
+            result = self.client.get(url)
+            return result.get('custom_attributes')
+        except Exception as e:
+            self.module.fail_json(msg="Failed to get {entity_type} custom attributes. Error: {error}".format(
+                entity_type=entity_type, error=e))
+
+    def add_custom_attributes(self, entity_type, entity_id, custom_attributes):
+        """ Returns the added custom attributes """
+        try:
+            url = '{api_url}/{entity_type}/{id}/custom_attributes'.format(
+                    api_url=self.api_url,
+                    entity_type=ManageIQ.supported_entities[entity_type],
+                    id=entity_id)
+            result = self.client.post(url, action='add', resources=custom_attributes)
+            self.changed = True
+            return result['results']
+        except Exception as e:
+            self.module.fail_json(msg="Failed to add the custom attributes. Error: {}".format(e))
+
+    def update_custom_attribute(self, entity_type, entity_id, ca, ca_href):
+        """ Returns the updated custom attributes """
+        try:
+            url = '{api_url}/{entity_type}/{id}/custom_attributes'.format(
+                    api_url=self.api_url,
+                    entity_type=ManageIQ.supported_entities[entity_type],
+                    id=entity_id)
+            ca_object = {'name': ca['name'], 'href': ca_href, 'value': ca['value']}
+            result = self.client.post(url, action='edit', resources=[ca_object])
+            self.changed = True
+            return result['results']
+        except Exception as e:
+            self.module.fail_json(msg="Failed to update the custom attribute {ca_name}. Error: {error}".format(ca_name=ca['name'], error=e))
+
+    def add_or_update_custom_attributes(self, entity_type, entity_name, custom_attributes):
+        """ Adds custom attributes to an entity in manageiq or updates the
+        attributes in case already exists
+
+        Returns:
+            the added or updated custom attributes, whether or not a change
+            took place and a short message describing the operation executed
+        """
+        added, updated = [], []
+        message = ""
+        # check if entity with the type and name passed exists in manageiq
+        entity_id = self.find_entity_by_name(entity_type, entity_name)
+        if entity_id:  # entity exists
+            ca_list = eval(custom_attributes)
```

Comment:
is it python const syntax (`ast.literal_eval()`)?  json (`json` module)?
But ansible is natively YAML, instead of taking a string and parsing it, why not have user pass structured data?
[I haven't yet read the PR, this comment might be out of context]